snowpack/src/scan-imports.ts

```diff
@@ -208,55 +240,13 @@ export async function scanImports(cwd: string, config: SnowpackConfig): Promise<
         return null;
       }
 
-      switch (baseExt) {
-        // Probably a license, a README, etc
-        case '': {
-          return null;
-        }
-        // Our import scanner can handle normal JS & even TypeScript without a problem.
-        case '.js':
-        case '.jsx':
-        case '.mjs':
-        case '.ts':
-        case '.css':
-        case '.tsx': {
-          return {
-            baseExt,
-            expandedExt,
-            locOnDisk: filePath,
-            contents: await fs.promises.readFile(filePath, 'utf-8'),
-          };
-        }
-        case '.html':
-        case '.vue':
-        case '.svelte': {
-          const result = await fs.promises.readFile(filePath, 'utf-8');
-          // TODO: Replace with matchAll once Node v10 is out of TLS.
-          // const allMatches = [...result.matchAll(new RegExp(HTML_JS_REGEX))];
-          const allMatches: string[][] = [];
-          let match;
-          let regex = new RegExp(HTML_JS_REGEX);
-          if (baseExt === '.svelte' || baseExt === '.vue') {
-            regex = new RegExp(SVELTE_VUE_REGEX); // scan <script> tags, not <script type="module">
-          }
-          while ((match = regex.exec(result))) {
-            allMatches.push(match);
-          }
-          return {
-            baseExt,
-            expandedExt,
-            locOnDisk: filePath,
-            // match[2] is the code inside the <script></script> element
-            contents: allMatches
-              .map((match) => match[2])
-              .filter((s) => s.trim())
-              .join('\n'),
-          };
-        }
+      // Probably a license, a README, etc
+      if (baseExt === '') {
+        return null;
       }
 
       // If we don't recognize the file type, it could be source. Warn just in case.
-      if (!mime.lookup(baseExt)) {
+      if (!Object.keys(srcFileExtensionMapping).includes(baseExt) && !mime.lookup(baseExt)) {
```

Comment:
A bit awkward, but with the switch case gone, we had to populate this extension list somehow.

Should we `return null` in this block?